presto-spi/src/main/java/io/prestosql/spi/block/MapBlock.java

```diff
@@ -217,27 +222,39 @@ public int getPositionCount()
     @Override
     public long getSizeInBytes()
     {
-        if (sizeInBytes < 0) {
-            calculateSize();
+        if (keySizeInBytes < 0) {
+            keySizeInBytes = calculateSize(keyBlock);
+        }
+        if (valueSizeInBytes < 0) {
+            valueSizeInBytes = calculateSize(valueBlock);
+        }
+        return baseSizeInBytes + keySizeInBytes + valueSizeInBytes;
+    }
+
+    @Override
+    public long getLoadedSizeInBytes()
+    {
+        if (keySizeInBytes < 0) {
```

Comment:
is check for `keyBlock.isLoaded` missing?

Reply:
Key block is always loaded in a map